### PR TITLE
[server] Add redundant filter on per message logging

### PR DESCRIPTION
**Pull request by majisourav99**
## Add redundant filter on per message logging

If ingestion task errors out or topic got unsubscribed, per message logging floods the log files. This PR adds some redundant checker to it.

## How was this PR tested?

CI
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java [hunk 1]
```diff
@@ -261,7 +261,11 @@ public void setChunkingInfo(
         record.setValueManifest(chunkedValueManifest);
         record.setRmdManifest(chunkedRmdManifest);
       } else {
-        LOGGER.error("Transient record is missing when trying to update value/RMD manifest.");
+        String msg = "Transient record is missing when trying to update value/RMD manifest for topic "
+            + ingestionTask.getKafkaVersionTopic() + " partition:" + subPartition;
+        if (partitionConsumptionState.isEndOfPushReceived() && !REDUNDANT_LOGGING_FILTER.isRedundantException(msg)) {
```
Comment (sixpluszero) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java:266]: nit: It is also okay to move the EOP check all the way up, but it is okay to keep it here as long as we don't over flow the log.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java [hunk 2]
```diff
@@ -261,7 +261,11 @@ public void setChunkingInfo(
         record.setValueManifest(chunkedValueManifest);
         record.setRmdManifest(chunkedRmdManifest);
       } else {
-        LOGGER.error("Transient record is missing when trying to update value/RMD manifest.");
+        String msg = "Transient record is missing when trying to update value/RMD manifest for topic "
+            + ingestionTask.getKafkaVersionTopic() + " partition:" + subPartition;
```
Comment (sixpluszero) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java:265]: nit: let's add a space between "partition:" and subPartition.

**Review by sixpluszero — COMMENTED** [review]
Need to fix the unit test coverage.

**Review by sixpluszero — DISMISSED** [review]
Thanks for the fix!

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java (hunk 1 above)
Comment (FelixGV) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java:266]: I would prefer moving the EOP check up as you mentioned... no need to allocate the string if that check short-circuits.

File: clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallbackTest.java [hunk 3]
```diff
@@ -4,11 +4,7 @@
 import static org.mockito.ArgumentMatchers.anyInt;
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.Mockito.doAnswer;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
 import static org.testng.Assert.assertEquals;
```
Comment (FelixGV) [inline clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallbackTest.java:8]: Can we revert that?

**Review by FelixGV — COMMENTED** [review]
Thanks!

File: clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallbackTest.java (hunk 3 above)
Comment (majisourav99) [reply clients/da-vinci-client/src/test/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallbackTest.java]: darn these intellij resetting setting on restart.

**Review by FelixGV — APPROVED** [review]
Looks good! Thanks!

**Review by FelixGV — APPROVED** [review]

**Pull request merged**